### PR TITLE
Add order sources in Order View page

### LINKED ISSUE #18966: Forgotten block inside migrated Order page : Sources
#### Describe the bug

When we migrated Order page, we forgot (because we did not it existed) the "Sources" block at the bottom of the page

#### Steps to Reproduce

1. Make an order on FO
2. See how it looks on legacy Order page (for example on 1.7.6), see the "sources" block at the bottom (attached is screenshot)
3. Compare to migrated page

**Screenshots**

Capture d’écran 2020-05-04 à 18 06 23

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add order sources in Order View page
| Type?         | new feature
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/18966
| How to test?  | See ticket

This is how it looks after migration:
Capture d’écran 2020-06-10 à 16 19 47

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19697)

### CODE REVIEW

**Comment by matks** [comment]
We have very few information about how this block is supposed to behave so I did what I thought relevant about rendering and markup.

@MatShir We might want to add a limit to how many items are displayed 🤔 else some orders could have hundred of elements in this bottom block

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ *#}
+
+  {% if orderForViewing.sources.sources is not empty %}
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig:26]: `orderForViewing.sources.sources` isn't redundant? :thinking: 
```suggestion
  {% if orderForViewing.sources.all is not empty %}
```
wdyt?

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                  <a href="{{ source.httpReferer }}">{{ source.httpReferer }}</a>
+                {% else %}
+                  -
+                {% endif %}
+                <br/>
+                <b>{{ 'To'|trans({}, 'Admin.Orderscustomers.Feature') }}</b>
+                <a href="http://{{ source.requestUri }}">{{ source.requestUri | slice(0,100) }}</a>
+                <br/>
+                {% if source.keywords != '' %}
+                  <b>{{ 'Keywords'|trans({}) }}</b>
```
Comment (Junebyun) [inline src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig]:
```suggestion
                  <b>{{ 'Keywords'|trans({}, 'Admin.Orderscustomers.Feature') }}</b>
```

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig (hunk 1 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig:26]: `orderForViewing.sources` is a collection class OrderSourcesForViewings which has a single property: an array of OrderSourceForViewing object, that I named very simply `$sources`

Both name are fine to me

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig (hunk 2 above)
Comment (matks) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig]: FYI in previous page, this word was without domain ; this is why I kept it like this. But If you think it's better to use the domain now, let's do it

**Review by PierreRambaud — DISMISSED** [review]

File: src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig (hunk 2 above)
Comment (Junebyun) [reply src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/View/sources.html.twig]:
```suggestion
                  <b>{{ 'Keywords'|trans({}, 'Admin.Global') }}</b>
```

@matks in this case I'll put in _Admin.Global_ instead! Thank you :)

**Review by Junebyun — DISMISSED** [review]

File: src/Core/Domain/Order/QueryResult/OrderSourceForViewing.php [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
```
Comment (Progi1984) [inline src/Core/Domain/Order/QueryResult/OrderSourceForViewing.php:25]:
```suggestion
 */
declare(strict_types=1);
```

File: src/Core/Domain/Order/QueryResult/OrderSourcesForViewing.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
```
Comment (Progi1984) [inline src/Core/Domain/Order/QueryResult/OrderSourcesForViewing.php:25]:
```suggestion
 */
declare(strict_types=1);
```

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @matks ! It's almost good for me, there is just the "items counter" at the top that is missing (the "è" is the screenshot below from legacy page) : 

https://user-images.githubusercontent.com/3830050/80987312-2ba58c00-8e32-11ea-8cf0-9c193a1788cb.png

**Comment by matks** [comment]
> https://user-images.githubusercontent.com/3830050/80987312-2ba58c00-8e32-11ea-8cf0-9c193a1788cb.png

I was hoping you would not notice :trollface:

**Review by Junebyun — DISMISSED** [review]

**Comment by matks** [comment]
Fixed code review feedbacks, Qa feedbacks and wording review feedbacks in https://github.com/PrestaShop/PrestaShop/pull/19697/commits/c8457e62fcb5c783d1b55123ae28a5ef83cd247f

It looks like this @Robin-Fischer-PS now:
Capture d’écran 2020-06-15 à 15 08 45

**Review by Junebyun — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Thanks @matks , you should know QA notices everything :stuck_out_tongue: 

It's OK !

**Comment by Progi1984** [comment]
Thanks @matks

**Pull request merged**